packages/backend/src/storage.ts

```diff
@@ -11,36 +11,187 @@ export default () => {
+            !Array.isArray(messageBatch) ||
+            !encryptedContactName
+        ) {
+            res.status(400).send('invalid schema');
+            return;
+        }
 
-            await req.app.locals.db.setUserStorageChunk(
+        try {
+            const ensName = normalizeEnsName(req.params.ensName);
+
+            await req.app.locals.db.storage_addMessageBatch(
                 ensName,
-                req.params.key,
-                stringify(req.body)!,
+                encryptedContactName,
+                messageBatch,
             );
+            return res.send();
+        } catch (e) {
+            return res.status(400).send('unable to add message batch');
+        }
+    });
 
-            res.json({
-                timestamp: new Date().getTime(),
-            });
+    router.get('/new/:ensName/getMessages/', async (req, res, next) => {
+        const pageNumber = parseInt(req.body.page);
+        const encryptedContactName = req.body.encryptedContactName as string;
+
+        if (isNaN(pageNumber) || !encryptedContactName) {
+            res.status(400).send('invalid schema');
+            return;
+        }
+        try {
+            const ensName = normalizeEnsName(req.params.ensName);
+            const messages = await req.app.locals.db.storage_getMessages(
+                ensName,
+                encryptedContactName,
+                pageNumber,
+            );
+            return res.json(messages);
+        } catch (e) {
+            next(e);
+        }
+    });
+    router.get('/new/:ensName/getNumberOfMessages/', async (req, res, next) => {
+        const encryptedContactName = req.body.encryptedContactName as string;
+
+        if (!encryptedContactName) {
+            res.status(400).send('invalid schema');
+            return;
+        }
+        try {
+            const ensName = normalizeEnsName(req.params.ensName);
+            const messages =
+                await req.app.locals.db.storage_getNumberOfMessages(
+                    ensName,
+                    encryptedContactName,
+                );
+            return res.json(messages);
+        } catch (e) {
+            next(e);
+        }
+    });
+
+    router.post('/new/:ensName/addConversation', async (req, res, next) => {
+        const { encryptedContactName } = req.body;
+        if (!encryptedContactName) {
+            res.status(400).send('invalid schema');
+            return;
+        }
+        try {
+            const ensName = normalizeEnsName(req.params.ensName);
+            const success = await req.app.locals.db.storage_addConversation(
+                ensName,
+                encryptedContactName,
+            );
+            if (success) {
+                return res.send();
+            }
+            res.status(400).send('unable to add conversation');
+        } catch (e) {
+            next(e);
+        }
+    });
+    router.get('/new/:ensName/conversationList', async (req, res, next) => {
```

Comment:
?

Reply:
the naming "conversationList" does not conform to the otherwise employed, descriptive approach. Therefore I propose to change the path to "getConversationList". 
I chose to express this in [sed-style](https://www.gnu.org/software/sed/manual/html_node/The-_0022s_0022-Command.html), forgot to include the trailing / though, which likely caused the confusion.